x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -196,12 +196,12 @@ public static StepKey getCurrentStepKey(Settings indexSettings) {
         }
     }
 
-    static Step getCurrentStep(PolicyStepsRegistry stepRegistry, String policy, Index index, Settings indexSettings) {
+    static Step getCurrentStep(PolicyStepsRegistry stepRegistry, String policy, IndexMetaData indexMetaData, Settings indexSettings) {
```

Comment:
This (among other things) is going to cause a merge conflict with #33783. Not sure which side that should be handled on, but fair warning.